src/main/java/au/org/emii/ncdfgenerator/NcdfDefinitionXMLParser.java

```diff
@@ -1,276 +1,374 @@
 
 package au.org.emii.ncdfgenerator;
 
-import au.org.emii.ncdfgenerator.IVariableEncoder;
-import au.org.emii.ncdfgenerator.VariableEncoder;
 
+import java.util.List;
 import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.Map;
-import java.util.AbstractMap.SimpleImmutableEntry;
 
+import java.sql.Connection;
+import java.io.InputStream;
+
+import org.w3c.dom.Document;
 import org.w3c.dom.Node;
 import org.w3c.dom.NamedNodeMap;
 
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.DocumentBuilder;
+
+import java.lang.Exception ;
+
+import au.org.emii.ncdfgenerator.IDimension;
+import au.org.emii.ncdfgenerator.NcdfGeneratorException;
 
-class NcdfDefinitionXMLParser
-{
-	private boolean isNodeName( Node node, String name )
-	{
-		return node.getNodeType() == Node.ELEMENT_NODE
-			&& node.getNodeName().equals( name );
-	}
 
-	private String nodeVal( Node node  )
+class Helper
+{
+	static String nodeVal( Node node )
 	{
 		Node child = node.getFirstChild();
 		if( child != null && child.getNodeType() == Node.TEXT_NODE )
 			return child.getNodeValue();
-
 		return "";
 	}
 
-
-	private Map< String, String> parseKeyVals( Node node )
+	static boolean isElementNode( Node node )
 	{
-		// have another version that does this for attributes
-		Map< String, String> m = new HashMap< String, String>();
-		for( Node child : new NodeWrapper(node) )
-			if( child.getNodeType() == Node.ELEMENT_NODE )
-				m.put( child.getNodeName(), nodeVal( child ) );
-
-		// add any xml attributes
-		NamedNodeMap attrs = node.getAttributes();
-		for( int i = 0; i < attrs.getLength(); ++i )
-		{
-			Node child = attrs.item( i) ;
-			m.put( child.getNodeName(), nodeVal( child ) );
-		}
-
-		return m;
+	    return node.getNodeType() == Node.ELEMENT_NODE;
 	}
 
-	private IDimension parseDimension( Node node)
+/*
+	static boolean isTextNode( Node node )
 	{
-		if( isNodeName( node, "dimension")) {
-			Map< String, String> m = parseKeyVals( node );
-			return new DimensionImpl( m.get( "name" ) );
-		}
-		return null;
+	    return node.getNodeType() == Node.TEXT_NODE;
 	}
 
-	// having a simple parse key-vals function, means we can do it with attributes as alternative syntax.
+*/
+}
+
 
-	private Map< String, IDimension> parseDimensions( Node node )
+class NcdfDefinitionXMLParser
+{
+	class Context
```

Comment:
I prefer reading from higher level of abstraction to lower level of abstraction except where conventions mandate the reverse (i.e. fields).  The initially interesting stuff (i.e. what the class is doing) is at the bottom of this class!  Moving these utility classes to separate files will make this easier to read as the parse method will be pretty much the first and only thing in this file.